derive_builder/tests/compile-fail/vis_conflict.rs

```diff
@@ -0,0 +1,31 @@
+#[macro_use]
+extern crate derive_builder;
+
+#[derive(Builder)]
+#[builder(public, vis = "pub(crate)")]
+pub struct Example {
+    // KNOWN ISSUE: The span for this error points to the macro invocation,
+    // rather than pointing to `public` or `private`.
+    #[builder(public, private)]
+    field: String,
+}
+
+// KNOWN ISSUE: There is no error for the struct-level conflicting visibility
+// declarations, only for build_fn. This is happening because `from_meta` for
+// `build_fn` returns `Err`, so the conflict check for the root `Options` does
+// not occur.
```

Comment:
I'm not thrilled with all these known issues, and it seems annoying that we refuse to emit anything in the case of ambiguous visibility (creating lots of secondary errors to deal with).

Could we instead defer conflict checking to after `FromDeriveInput::from_derive_input` has returned `Ok`, putting it in a new `check_nonfatal_errors` method, and have `as_expressed_vis` return the most conservative of the passed-in visibilities? That would mean we produce errors for all collisions, regardless of location, and that we still emit a builder struct.

Reply:
As an interim approach here, I've deferred the validation but left the panic in place. This makes it impossible for invalid cases to produce working code, and since I almost missed one validation path during development I think that's better than trying to continue past the issue for now.